e2e/tests/dashboard-tests.spec.ts

```diff
@@ -0,0 +1,150 @@
+import { expect, test } from "@playwright/test";
+import { startApp } from "../utils/core-e2e-utils";
+
+async function setFixedDate(page, fixedDate) {
+  const fakeNow = new Date(fixedDate).valueOf(); // Convert the fixed date to a timestamp
+  await page.addInitScript(`{
+    Date = class extends Date {
+      constructor(...args) {
+        if (args.length === 0) {
+          super(${fakeNow}); // Default to the fixed date
+        } else {
+          super(...args); // Handle explicitly provided dates
+        }
+      }
+    }
+    const __DateNowOffset = ${fakeNow} - Date.now(); // Offset for Date.now()
+    const __DateNow = Date.now;
+    Date.now = () => __DateNow() + __DateNowOffset; // Override Date.now()
+  }`);
+}
+
+test.describe("Dashboard Page Tests", () => {
+  test.beforeEach(async ({ page }, testInfo) => {
+    console.log(`Running test case - ${testInfo.title}`);
+
+    // Set fixed date before navigating to the app
+    await setFixedDate(page, "1/1/2025");
+
+    await startApp(page);
+  });
+
+  test("Verify Quick Actions widget", async ({ page }) => {
+    // Check "Quick Actions" widget is visible
+    const quickActionsElement = page.locator("text=Quick actions");
+    await expect(quickActionsElement).toBeVisible();
+
+    // Verify "Record Attendance" button in Quick Actions
+    const recordAttendanceButton = page.getByRole("cell", {
+      name: "Record Attendance",
+    });
+    await expect(recordAttendanceButton).toBeVisible();
+  });
+
+  test("Verify children count is displayed", async ({ page }) => {
+    const childrenCount = page.locator("app-entity-count-dashboard-widget");
+    await expect(childrenCount).toContainText(/107/);
+  });
+
+  test("Verify Tasks Due widget and tasks", async ({ page }) => {
+    // Check "Tasks Due" widget is visible
+    const tasksDueElement = page.getByText("Tasks due");
+    await expect(tasksDueElement).toBeVisible();
+
+    // Verify at least one task is listed
+    const taskElements = page.locator("app-widget-content");
+    const taskCount = await taskElements.count();
+    expect(taskCount).toBeGreaterThan(0);
+  });
+
+  test("Navigate to Attendance page and verify sections", async ({ page }) => {
```

Comment:
The file here is called "Dashboard Tests". This case should probably be moved to the attendance tests file, if it is not already tested there somewhere in different tests?